pkg/cfn/builder/block_device_mapping.go

```diff
@@ -1,57 +1,79 @@
 package builder
 
 import (
-	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 	gfnec2 "github.com/weaveworks/goformation/v4/cloudformation/ec2"
 	gfnt "github.com/weaveworks/goformation/v4/cloudformation/types"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 )
 
 const rootDevice = "/dev/xvda"
 
 func makeBlockDeviceMappings(ng *api.NodeGroupBase) []gfnec2.LaunchTemplate_BlockDeviceMapping {
-	volumeSize := ng.VolumeSize
+	var mappings []gfnec2.LaunchTemplate_BlockDeviceMapping
+	mapping := makeBlockDeviceMapping(api.VolumeMapping{
+		VolumeSize:       ng.VolumeSize,
+		VolumeType:       ng.VolumeType,
+		VolumeName:       ng.VolumeName,
+		VolumeEncrypted:  ng.VolumeEncrypted,
+		VolumeKmsKeyID:   ng.VolumeKmsKeyID,
+		VolumeIOPS:       ng.VolumeIOPS,
+		VolumeThroughput: ng.VolumeThroughput,
+	})
+
+	if mapping != nil {
+		mappings = append(mappings, *mapping)
+	}
+
+	for _, volume := range ng.AdditionalVolumes {
+		mappings = append(mappings, *makeBlockDeviceMapping(volume))
+	}
```

Comment:
we only check if the `mapping` is nil for the default volume, not the additional volumes?

Reply:
That is a good thought, though I think if someone defined an additional volume they surely would set the size, right? But, maybe that's a problem in any case. Like, if additional volumes is defined, size should be required? WDYT?

Reply:
In any case I'll add the check and extend the test case to include one  maybe.

Reply:
perhaps we are missing checks in the `validation.go` file that that the minimum required fields are set?

Reply:
Possibly.

Reply:
Uh, it's because we just set the mapping field to nil if the size wasn't defined, never really doing anything with it. This is a bit weird. :D

Reply:
Aaaah, I'm not touching this. :D I'll add the nil check :D

Reply:
Oh shit, interesting. We have a thing `setVolumeDefaults` that sets default.

Reply:
OOOOH. Now I see.

Reply:
We always have these fields set. You can just overwrite them. 🤦 . So I need a different kind of check of the additional thing.